index.js

```diff
@@ -94,15 +99,49 @@ const getAddresses = () => {
   return _rlAddressToRequests
 }
 
+
+const hash = (address) => {
+  return Crypto.SHA256(address).toString(Crypto.enc.Hex);
+
+}
+
 const setTotalActiveUsers = (req) => {
-  const address = req.connection.remoteAddress
-  const lastRequestAt = _rlAddressToRequests[address] ? _rlAddressToRequests[address].lastRequestAt : null
+  const address = hash(req.connection.remoteAddress)
+  incrementActiveUsers(address)
+}
 
-  if(!lastRequestAt || lastRequestAt.diff(_lastActiveTimeout) < 0){ //The last connection was before we zeroed the totalActiveUsers field
+incrementActiveUsers = (address) => {
+  if(!userActive(address)){ //The last connection was before we zeroed the totalActiveUsers field
     _totalActiveUsers++
   }
 }
 
+decrementActiveUsers = (address) => {
+  if (userActive(address)) { //The last connection was after we zeroed the totalActiveUsers field
+    _totalActiveUsers--
+  }
+}
+
+const userActive = (address) => {  
+  const lastRequestAt = _rlAddressToRequests[address] ? _rlAddressToRequests[address].lastRequestAt : null
+  return (lastRequestAt && lastRequestAt.diff(_lastActiveTimeout) >= 0)
+}
+
+const removeUser = (address) => {
+  decrementActiveUsers(address) //If user is active we must decrement the active users field
+  delete _rlAddressToRequests[address]
+  
+}
+
+const addUser = (address, userTableMaxSize, now) => {
+  _userQueue.push(address)
+  if (Object.keys(_rlAddressToRequests).length >= userTableMaxSize) { //We have reached an overflow scenario
```

Comment:
The fact that we add an object and _then_ checks if it has overflown is a bit weird I think. We should first do the check "will this object fit, if not remove first in queue".

Reply:
Greater than or **equal** is also not an overflow scenario

Reply:
Move the push to function end. Greater than or equal is an overflow scenario. If we add one more(which we will). It will overflow.